flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCTableSourceSinkFactory.java

```diff
@@ -136,6 +146,35 @@
 		return builder.build();
 	}
 
+	/**
+	 * The original table schema may contain generated columns which shouldn't be produced/consumed
+	 * by TableSource/TableSink. And the original TIMESTAMP/DATE/TIME types uses LocalDateTime/LocalDate/LocalTime
+	 * as the conversion classes, however, JDBC connector uses Timestamp/Date/Time classes. So that
+	 * we bridge them to the expected conversion classes.
+	 */
+	private TableSchema getPhysicalTableSchema(DescriptorProperties descriptorProperties) {
+		TableSchema schema = descriptorProperties.getTableSchema(SCHEMA);
+		TableSchema.Builder physicalSchemaBuilder = TableSchema.builder();
+		schema.getTableColumns()
+			.forEach(c -> {
+				if (!c.isGenerated()) {
+					LogicalTypeRoot root = c.getType().getLogicalType().getTypeRoot();
+					final DataType type;
+					if (root == LogicalTypeRoot.TIMESTAMP_WITHOUT_TIME_ZONE) {
+						type = c.getType().bridgedTo(Timestamp.class);
```

Comment:
But not work for nested fields? right?

Reply:
You are right, currently JDBC connector only support atomic types as columns.

Reply:
I think we should have a method to bridgedTo all timestamp type. This can avoid potential bug.

Reply:
But this shouldn't be a method of TableSchema. This can be a util method, but I'm not sure where it belongs.